frontend/app/components/catList.jsx

```diff
@@ -1,27 +1,91 @@
 import React from 'react'
 import CatCard from './catCard'
+import { connect } from 'react-redux'
 import { Grid, Row } from 'react-bootstrap'
+import Infinite from 'react-infinite'
+import Media from 'react-media'
+import { fetchCatList } from '../actions/catListActions'
 
 class CatList extends React.Component {
   constructor(props) {
     super(props)
+    this.handleInfiniteLoad = this.handleInfiniteLoad.bind(this)
+    this.state = {
+      offset: 0,
+      isInfiniteLoading: false,
+    }
   }
 
+  handleInfiniteLoad(){
+    this.setState({isInfiniteLoading: true})
+    setTimeout(() => {
+      this.setState({ offset : this.state.offset + 30,
+                      isInfiniteLoading: false}),
+      this.props.fetchCatList(this.props.zipcode, this.state.offset)
+    }, 2500)
+  }
+
+  elementInfiniteLoad() {
+    return <div className="loading">
+            <h3>Loading...</h3>
+           </div>;
+}
   render() {
     const cats = this.props.data
-    const catItems = cats.map((cat, id) => {
-      return(
-        <CatCard cat={cat} key={cat.id} />
-      )
+    const groupSize = 3
+    const mobileRows = cats.map((cat, id) => {
+        return(
+            <CatCard cat={cat} key={cat.id} />
+        )
     })
+    const rows = mobileRows.reduce((row, element, index)=>{
+        index % groupSize === 0 && row.push([]);
+        row[row.length - 1].push(element);
+        return row
+    }, []).map((rowContent, index) => {
+      return <div className="row" key={index}>{rowContent}</div>
+    })
```

Comment:
this is not very readable. I understand the elegant feeling of packing a ton of logic into as few lines as possible, but I think you overdid it here. if you don't want to refactor for readability that's fine but at least add an explanatory comment